### PR TITLE
refactor(date-picker-month-header): tailwindify

### PR DESCRIPTION
**Pull request by julio8a**
Supports #1500

### CODE REVIEW

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss [hunk 1]
```diff
@@ -1,137 +1,142 @@
 .header {
-  display: flex;
-  justify-content: space-between;
+  @apply flex justify-between;
+  //FIXME: Can this just be 4px? px-1
```
Comment (paulcpederson) [inline src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:3]: yes, you need to also change the negative margin on the `chevron` to `-4px` as well

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss [hunk 2]
```diff
@@ -1,137 +1,142 @@
...
-  color: var(--calcite-ui-text-1);
-  background: var(--calcite-ui-foreground-1);
-  font-size: inherit;
-  font-weight: 500;
-  line-height: 1.25;
-  margin: 0 4px;
-  display: inline-block;
+  @apply text-color-1
+    bg-foreground-1
+    font-inherit
```
Comment (paulcpederson) [inline src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:97]: this is not doing what you want it to. We need `font-size: inherit` and this is `font-family: inherit`. The month text then inherits the browser default h2 and looks real bad: 

Screen Shot 2021-03-18 at 4 11 50 PM

**Comment by julio8a** [comment]
changes made on this one

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss [hunk 3]
```diff
@@ -1,137 +1,139 @@
...
+    @apply h-16;
   }
 }
 
 .chevron {
-  color: var(--calcite-ui-text-2);
-  flex-grow: 0;
+  @apply text-color-2
+    flex-grow-0
+    box-border
```
Comment (jrowlingson) [inline src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:36]: Intended?

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss [hunk 4]
```diff
@@ -1,137 +1,139 @@
...
+    outline-none
+    py-0
+    px-1
+    my-0
+    -mx-1
```
Comment (jrowlingson) [inline src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:41]: py-0 my-0 artifacts from the shorthand? remove?

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss [hunk 5]
```diff
@@ -1,137 +1,139 @@
...
-    @include focus-style-inset();
+    @apply focus-inset;
   }
 
   &:hover,
   &:focus {
-    fill: var(--calcite-ui-text-1);
-    background-color: var(--calcite-ui-foreground-2);
+    @apply bg-foreground-2;
+    fill: theme("textColor.1");
```
Comment (jrowlingson) [inline src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:61]: fill-color-1

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss [hunk 6]
```diff
@@ -1,137 +1,139 @@
...
 
   &:active {
-    background-color: var(--calcite-ui-foreground-3);
+    @apply bg-foreground-3;
   }
 
   &[aria-disabled="true"] {
-    pointer-events: none;
-    opacity: 0;
+    @apply pointer-events-none bg-opacity-0;
```
Comment (jrowlingson) [inline src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:69]: opacity-0

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss [hunk 7]
```diff
@@ -1,137 +1,139 @@
...
-  flex: 1 1 auto;
-  display: flex;
-  align-items: center;
-  justify-content: center;
-  flex-direction: row;
-  line-height: 1;
-  margin: auto 0;
-  text-align: center;
+  @apply flex-auto
+    flex-auto
```
Comment (jrowlingson) [inline src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:75]: Dedup

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss [hunk 8]
```diff
@@ -1,137 +1,139 @@
...
+    items-center
+    justify-center
+    flex-row
```
Comment (jrowlingson) [inline src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:78]: needed?

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss [hunk 9]
```diff
@@ -1,137 +1,139 @@
...
+    leading-none
+    my-auto
+    mx-0
```
Comment (jrowlingson) [inline src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:81]: needed?

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss [hunk 10]
```diff
@@ -1,137 +1,139 @@
...
 .month,
 .year,
 .suffix {
-  color: var(--calcite-ui-text-1);
-  background: var(--calcite-ui-foreground-1);
+  @apply text-color-1
+    bg-foreground-1
+    font-normal
+    leading-tight
+    my-0
```
Comment (jrowlingson) [inline src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:96]: shorthand artifact et al?

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss [hunk 11]
```diff
@@ -1,137 +1,139 @@
...
+    border-none
+    w-12
+    p-0
+    bg-transparent
+    relative
+    focus-base;
   z-index: 2;
-  @include focus-style-base();
   &:hover {
     transition: outline-color 100ms ease-in-out;
```
Comment (jrowlingson) [inline src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:113]: `transition-property: outline-color` and `duration-100 ease-in-out`

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss [hunk 12]
```diff
@@ -1,137 +1,139 @@
...
     outline: 2px solid var(--calcite-ui-border-2);
     outline-offset: 2px;
```
Comment (jrowlingson) [inline src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:115]: This is close to the outline- default. Lets add this to the list.

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss [hunk 13]
```diff
@@ -1,137 +1,139 @@
...
-  left: 0;
+  @apply absolute
+    w-16
+    whitespace-no-wrap
+    text-left
+    top-0
+    left-0;
 }
 
 .suffix__invisible {
```
Comment (jrowlingson) [inline src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:139]: invisible

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss (hunk 3 above)
Comment (julio8a) [reply src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:36]: no, fixed.

File: src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss (hunk 4 above)
Comment (julio8a) [reply src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.scss:41]: removed

**Comment by julio8a** [comment]
This one is ready for review.

**Review by paulcpederson — APPROVED** [review]

**Comment by paulcpederson** [comment]
looks good.

**Pull request merged**